### PR TITLE
[Checkout coverage] Place order concept

### PR DESCRIPTION
**Pull request by rogyar**
### Description (*)
This PR introduces a new mutation for placing an order. In order to place the order, the quote (cart) should contain a product as well as shipping/billing data assigned. 

### Fixed Issues (if relevant)
1. #58 : [Checkout coverage] Place order
2. #295 : [Place order] Place order mutation

### Manual testing scenarios (*)
- Add a product to the shopping cart
- Assign a shipping method and shipping address 
- Assign a payment method and billing address
- Use the following mutation for placing the order:

```graphql
mutation {
  placeOrder(input: {cart_id: "$maskedQuoteId"}) {
    order {
      order_id
    }
  }
}
```

### CODE REVIEW

File: app/code/Magento/QuoteGraphQl/Model/Resolver/PlaceOrder.php [hunk 1]
```diff
@@ -0,0 +1,87 @@
...
+                'order' => [
+                    'order_id' => $order->getIncrementId()
+                ]
+            ];
+        } catch (LocalizedException $exception) {
+            throw new GraphQlInputException(
+                __('Unable to place order: %message', ['message' => $exception->getMessage()])
+            );
+        } catch (\Exception $exception) {
+            throw new GraphQlInputException(
```
Comment (naydav) [inline app/code/Magento/QuoteGraphQl/Model/Resolver/PlaceOrder.php]: We don't need to show all of error messages to client (just Localized)

File: dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/PlaceOrderTest.php [hunk 2]
```diff
@@ -0,0 +1,224 @@
...
+        $query = $this->preparePlaceOrderQuery();
+
+        self::expectExceptionMessageRegExp('/The current user cannot perform operations on cart*/');
+        $this->graphQlQuery($query);
+    }
+
+    /**
+     * @magentoApiDataFixture Magento/Checkout/_files/quote_with_check_payment.php
+     */
+    public function testPlaceOrderWithOutOfStockProduct()
```
Comment (TomashKhamlai) [inline dev/tests/api-functional/testsuite/Magento/GraphQl/Quote/PlaceOrderTest.php]: Does it cover the situation here #437 Screenshot 2?
I believe it doesn't even if problem doesn't appear here. Actually I haven't tested "Place order concept" yet

**Comment by galaoleksandr** [comment]
Hello. I cannot place order on guest cart when shipping and payment methods are set on front.

**Comment by rogyar** [comment]
Hi @galaoleksandr. Could you provide more information about the error message you are getting? 
Also, could you describe, please, how you set payment method on the storefront without placing an order? Technically, when you click on a payment method radio button (on checkout page) the payment method is not being assigned to the quote. It's only being saved for user session on the storefront, so only customer's browser knows about the selected option:
image

In case of shipping method, it's being submitted to the backend once you go from shipping to payment step. The payment method submitted to the backend only after placing an order

Thank you

**Comment by galaoleksandr** [comment]
Hi @rogyar. I've tried to set payment method using GraphQl Playground. Now I get the next error: 
image

**Comment by rogyar** [comment]
Hi @galaoleksandr. Were you able to place an order (I see the label removed)? I can provide you with detailed steps.

Thank you

**Pull request merged**